prime-router/src/main/kotlin/transport/RESTTransport.kt

```diff
@@ -90,7 +91,25 @@ class RESTTransport(private val httpClient: HttpClient? = null) : ITransport {
         val receiver = header.receiver ?: error("No receiver defined for report $reportId")
         val reportContent: ByteArray = header.content ?: error("No content for report $reportId")
         // get the file name, or create one from the report ID, NY requires a file name in the POST
-        val fileName = header.reportFile.externalName ?: "$reportId.hl7"
+        val fileName = if (restTransportInfo.headers["File-Name"] != null) {
```

Comment:
Why can we not use the existing function inside Report.kt:formFilename to build the filename that uses the receiver setting fileFormat and the file-name-templates.yml?

Reply:
Thank you, I used the utility to generate the fileName.